Removing ElasticSearch Brew instructions

cmgorton opened the pull request:
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description
If you try to install brew as directed you get this Error:
Error: Calling Installation of elasticsearch-oss from a GitHub commit URL is disabled! Use 'brew extract elasticsearch-oss' to stable tap on GitHub instead.

Brew/ElasticSearch no longer allows you to tap as we previously have suggested in the doc. Also we can no longer grab the 7.5.2 version with brew.

We have decided to remove the instructions for now until we upgrade ElasticSearch if possible.
## Related Tickets & Documents

## QA Instructions, Screenshots, Recordings
![Screen Shot 2020-09-17 at 2 10 18 PM](https://user-images.githubusercontent.com/21039864/93522988-8de9bf80-f8ef-11ea-82d2-2bb26faa218d.png)
## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.forem.com
- [ ] readme
- [ ] no documentation needed

nickytonline commented on docs/installation/mac.md:
```diff
@@ -12,15 +12,16 @@ title: macOS
 2. With the Ruby version manager, install the Ruby version listed on our badge.
-   (i.e. with rbenv: `rbenv install $(cat .ruby-version)`)
+   (i.e. with rbenv: `rbenv install *latest badge version*`) 
+    example: `rbenv install 2.7.1`
```
``(i.e. with rbenv: `rbenv install $(cat .ruby-version)`)`` should stay as it is reading the `.ruby-version` file from the forked Forem repository.

If this happens before the docs mention grabbing the repository, we should rework that too.

cmgorton replied:
Yes it is happening before there is mention of forking the repo but we can mention to fork the repository first instead and keep `(i.e. with rbenv: `rbenv install $(cat .ruby-version)`)`

nickytonline reviewed (commented):
Looks good, just one comment.

rhymes commented on docs/installation/mac.md:
```diff
@@ -61,11 +62,10 @@ redis-cli ping
 ### Elasticsearch
 
-Forem requires a version of Elasticsearch between 7.1 and 7.5. Version 7.6 is
+Forem requires a version of Elasticsearch between 7.1 and 7.5 to be running. Version 7.6 is
 not supported. We recommend version 7.5.2.
```
I think we should update this. In production we run 7.5.2 (though it technically runs with 7.9 as well but it's not officially supported), we don't really know if it still works on 7.1 to 7.4.

Maybe we should mention 7.5 or higher, like we do for PostreSQL?

rhymes commented on docs/installation/mac.md:
````diff
@@ -107,25 +107,6 @@ To start elasticsearch as a daemonized process:
 ./bin/elasticsearch -d
 ```
 
-### Installing Elasticsearch with Homebrew
````
If we switch to "it runs on 7.5 or higher" we could in theory keep these instructions, after all the official docs still support ES via Homebrew: https://www.elastic.co/guide/en/elasticsearch/reference/current/brew.html

(I'm using `elasticsearch-oss` 7.9.1 via `brew tap elastic/tap` in development)

The risk is that someone will install ES 8 when it comes out and maybe it won't work so I'm also in favor of scrapping them.

If we keep them I'd replace them with something like:

```
brew tap elastic/tap
brew install elastic/tap/elasticsearch-oss
brew pin elasticsearch-oss
```

The rest of the post installation instructions would be the same.

joshpuetz replied:
Given my recent conversations with @mstruve and @jdoss, it seems like "unofficially" supporting 7.5 and above is a good step. I'd MUCH rather keep in the homebrew installation instructions if we can!

(@cmgorton and I went 10 rounds with `brew` yesterday and there's just no good/easy way to install older versions of Elasticsearch anymore)

mstruve replied:
I'm cool with bumping it to 7.x version that works.

cmgorton replied:
Ok, maybe I should close this PR for now and create a Draft PR until we all come to a consensus on brew and Elasticsearch. Thoughts?

joshpuetz replied:
I'd be comfortable just changing this PR to a) mentioned we use Elasticsearch 7.5 and up, and b) changing the home-brew instructions to pull right from the latest version like @rhymes mentioned at https://github.com/forem/forem/pull/10370#discussion_r490709809!

cmgorton replied:
Hey I am circling back on this as I know I let it lapse. I had a community member who was going to pair with Nick this week unable to get to get the installation working because of elasticsearch. Should I recommend he pull from the latest version like @rhymes and @joshpuetz mentioned?

If we are still ok with that I can also update this PR to do the same.

joshpuetz replied:
I'd vote for specifying the latest version of Elasticsearch!

mstruve replied:
Any 7.x version of Elasticsearch will work and can be recommended

rhymes reviewed (commented):
Great find @cmgorton, seems like our instructions are a bit outdated :( 

Left notes about how we could avoid scraping all the homebrew instructions by instead updating them

cmgorton commented:
I made updates per everyones recommendation:

Added back line 15 per Nicks recommendation  _(i.e. with rbenv: `rbenv install $(cat .ruby-version)`)_

Line 65  updated to  7.5 or higher for Elasticsearch per Rhymes recommendation
_Forem requires a version of Elasticsearch 7.5 or higher to be running.  We recommend version 7.5.2._

Line 110 update:
Added back in the Homebrew instructions but updated commands to `install elastic/tap/elasticsearch-oss`

mstruve commented on docs/installation/mac.md:
```diff
@@ -61,11 +61,9 @@ redis-cli ping
 
-Forem requires a version of Elasticsearch between 7.1 and 7.5. Version 7.6 is
-not supported. We recommend version 7.5.2.
+Forem requires a version of Elasticsearch 7.5 or higher to be running. We recommend version 7.5.2.
```
```suggestion
Forem requires Elasticsearch 7.x to run. We recommend version 7.5.2.
```

mstruve commented on docs/installation/mac.md:
```diff
@@ -38,7 +38,7 @@ You can install ImageMagick with `brew install imagemagick`.
 ### Redis
 
-Forem requires Redis version 4.0 or higher.
+Forem requires Redis version 4.0 or higher to be running.
```
```suggestion
Forem requires Redis version 4.x to run.
```

cmgorton replied:
Yeah, I like this suggestions! Thank you Molly.

cmgorton replied:
Updated with these suggestions along with the section on PostgreSQL that also had "to be running." Thanks for the suggestions @mstruve

mstruve reviewed (approved):
LGTM! I left a couple of wording suggestions for the versions to keep our major versions consistent but otherwise I like it!
